erpnext/hr/doctype/shift_request/shift_request.py

```diff
@@ -4,7 +4,83 @@
 
 from __future__ import unicode_literals
 import frappe
+from frappe import _
 from frappe.model.document import Document
+from frappe.utils import cint, cstr, date_diff, flt, formatdate, getdate
+
+class OverlapError(frappe.ValidationError): pass
 
 class ShiftRequest(Document):
-	pass
+	def validate(self):
+		self.validate_dates();
+		self.validate_shift_request_overlap_dates();
+
+	def on_submit(self):
+		date_list = self.get_date_list(self.from_date, self.to_date)
+		for date in date_list:
+			assignment_doc = frappe.new_doc("Shift Assignment")
+			assignment_doc.company = self.company
+			assignment_doc.shift_type = self.shift_type
+			assignment_doc.employee = self.employee
+			assignment_doc.date = date
+			assignment_doc.shift_request = self.name
+			assignment_doc.insert()
+			assignment_doc.submit()
+
+	def validate_dates(self):
+		if self.from_date and self.to_date and (getdate(self.to_date) < getdate(self.from_date)):
+			frappe.throw(_("To date cannot be before from date"))
+
+	def validate_shift_request_overlap_dates(self):
+			if not self.name:
+				self.name = "New Shift Request"
+
+			for d in frappe.db.sql("""
+				select
+					name, shift_type, from_date, to_date
+				from `tabShift Request`
+				where employee = %(employee)s and docstatus < 2
+				and to_date >= %(from_date)s and from_date <= %(to_date)s
+				and name != %(name)s""", {
+					"employee": self.employee,
+					"from_date": self.from_date,
+					"to_date": self.to_date,
+					"name": self.name
+				}, as_dict = 1):
+
+				if (((getdate(self.from_date) >= getdate(d.to_date) 
+					or getdate(self.from_date) <= getdate(d.from_date))
+					or (getdate(self.to_date) >= getdate(d.from_date) 
+					or getdate(self.to_date) <= getdate(d.to_date))) and
+					self.shift_type == d.shift_type ):
+						self.throw_overlap_error(d)
+
+	def throw_overlap_error(self, d):
+		msg = _("Employee {0} has already applied for {1} between {2} and {3} : ").format(self.employee,
+			d['shift_type'], formatdate(d['from_date']), formatdate(d['to_date'])) \
+			+ """ <b><a href="#Form/Shift Request/{0}">{0}</a></b>""".format(d["name"])
+		frappe.throw(msg, OverlapError)
+
+	def get_date_list(self, start_date, end_date):
```

Comment:
get_working_days ?

Reply:
@nabinhait , made the other review comment changes. Related to this, may need some more clarification.

Reply:
Renaming the function to `get_working_dates` will clear the objective of the function.